lib/storage.js

```diff
@@ -209,6 +209,60 @@ exports.set = function(key, json, callback) {
 };
 
 /**
+ * @summary Write multiple sets of user data
+ * @function
+ * @public
+ *
+ * @param {Object} keys
+ * @param {Function} callback - callback (error, done)
+ *
+ * @example
+ * const storage = require('electron-json-storage');
+ *
+ * storage.setMany({ foo: 'bar', bar: 'foo' }, function(error, done) {
+ *  if (error) {
+ *    console.log('Error saving ' + error.key);
+ *    throw error.error;
+ *  }
+ *  if (finished) console.log('finished');
+ * });
+ */
+exports.setMany = function(keys, callback) {
+  if (!keys || keys.constructor !== Object) {
+    callback(new Error('Invalid keys'), true);
+    return;
+  }
+
+  const entries = _.map(keys, function(value, key) {
+    return { key: key, value: value };
+  });
+
+  function process() {
+    const entry = entries.shift();
+    const key = entry.key;
+    const value = entry.value;
+
+    exports.set(key, value, function(error) {
+      if (error) {
+        callback({ error: error, key: key }, entries.length === 0);
```

Comment:
I see. I think we should return a list of failed keys along with their respective errors, and only call the final callback once.